provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/client/RestVlerClient.java

```diff
@@ -0,0 +1,171 @@
+package gov.va.api.health.providerdirectory.service.client;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import gov.va.api.health.providerdirectory.service.AddressResponse;
+import gov.va.api.health.providerdirectory.service.VlerResponse;
+import java.nio.charset.Charset;
+import java.security.cert.X509Certificate;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.concurrent.Callable;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import javax.net.ssl.SSLContext;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/** REST implementation of PPMS client. */
+@Slf4j
+@Component
+public class RestVlerClient implements VlerClient {
+
+  private final String baseUrl;
+
+  private final RestTemplate restTemplate;
+
+  @Value("${vler.pubkey}")
+  String pubKey;
+
+  @Value("${vler.prvkey}")
+  String prvKey;
+
+  @Value("${vler.truststore.password}")
+  String truststorePassword;
+
+  @Value("${vler.truststore.location}")
+  String truststoreLocation;
+
+  @Value("${vler.url}")
+  String vlerUrl;
+
+  public RestVlerClient(
+      @Value("${vler.url}") String baseUrl, @Autowired RestTemplate restTemplate) {
```

Comment:
For the sake of consistency, let's add all of the fields to this constructor and move the `@Value()` annotations down here.